build.sbt

```diff
@@ -0,0 +1,30 @@
+// shadow sbt-scalajs' crossProject and CrossType until Scala.js 1.0.0 is released
+import sbtcrossproject.{crossProject, CrossType}
+
+inThisBuild(Def.settings(
+  crossScalaVersions := Seq("2.10.7", "2.11.12", "2.12.6"),
+  scalaVersion := crossScalaVersions.value.last,
+  version := "0.1.0-SNAPSHOT",
+
+  scalacOptions ++= Seq(
+    "-deprecation",
+    "-feature",
+    "-encoding",
+    "utf-8",
+    "-Xfatal-warnings",
+  )
+))
+
+lazy val reflect = crossProject(JSPlatform, JVMPlatform)
+  .in(file("."))
+  .settings(
+    scalacOptions in (Compile, doc) -= "-Xfatal-warnings",
+  )
+  .jvmSettings(
+    compileOrder := CompileOrder.JavaThenScala,
```

Comment:
Why this?

Reply:
The default is mixed compilation, where scalac is invoked with the .java source files on its "classpath", and so scalac will compile the API of the .java files to use in the .scala files. After, the .java files are sent to javac with the compiled .class files produced by scalac. This allows both Scala to "see" Java and Java to see Scala. It's a bit slow, though.

In our case, we don't need any Java class to see anything from Scala, so forcing Java-then-Scala is more efficient: first javac is invoked for .java files, then scalac is invoked for .scala files with the .class files produced by javac on its classpath.

tl;dr it's not *necessary*, but it improves compile time.